python/ray/autoscaler/v2/tests/test_e2e.py

```diff
@@ -315,6 +315,80 @@ def verify():
         cluster.shutdown()
 
 
+def test_non_corrupted_resources():
+    """
+    Test that when node's local gc happens due to object store pressure,
+    the message doesn't corrupt the resource view on the gcs.A
+    See issue https://github.com/ray-project/ray/issues/39644
+    """
+    num_worker_nodes = 5
+    cluster = AutoscalingCluster(
+        head_resources={"CPU": 2, "object_store_memory": 100 * 1024 * 1024},
+        worker_node_types={
+            "type-1": {
+                "resources": {"CPU": 2},
+                "node_config": {},
+                "min_workers": num_worker_nodes,
+                "max_workers": num_worker_nodes,
+            },
+        },
+    )
+
+    driver_script = """
+
+import ray
+import time
+
+ray.init("auto")
+
+@ray.remote(num_cpus=1)
+def foo():
+    ray.put(bytearray(1024*1024* 50))
+
+
+while True:
+    ray.get([foo.remote() for _ in range(50)])
+"""
+
+    try:
+        # This should trigger many COMMANDS messages from TryGlobalGC on GCS.
```

Comment:
actually don't we want many COMMANDS message from raylet?

Reply:
oh yes, we want those to be triggered - thus all the sys configs.

Reply:
Does that mean the comment is not accurate?

Reply:
i guess i do wanna trigger many COMMANDS here? i might be missing something.